process/block/shardblock.go

```diff
@@ -905,16 +928,32 @@ func (sp *shardProcessor) CommitBlock(
 
 	sp.displayPoolsInfo()
 
+	timing = append(timing, time.Now())
 	errNotCritical = sp.removeBlockDataFromPools(headerHandler, bodyHandler)
 	if errNotCritical != nil {
 		log.Debug("removeBlockDataFromPools", "error", errNotCritical.Error())
 	}
 
+	timing = append(timing, time.Now())
 	sp.cleanupPools(headerHandler)
+	timing = append(timing, time.Now())
+
+	logTimings(timing)
 
 	return nil
 }
 
+func logTimings(timings []time.Time) {
+	totalDuration := timings[len(timings)-1].UnixNano() - timings[0].UnixNano()
```

Comment:
same as found in #2214 
we have something for these types of measurements. Might take a look in core/stopWatch.go. It supports identifiers and is optimized to return the values as []interface{} to match the input for a log print.